compat/src/jsx.d.ts

```diff
@@ -0,0 +1,4 @@
+import { JSXInternal } from '../../src/jsx';
+
+export = JSXInternal;
+export as namespace JSX;
```

Comment:
I think this broke typescript before

https://github.com/preactjs/preact-cli/issues/1030

Reply:
In what way? I skimmed through the referenced issue but did not understand how this breaks typescript. This file declares the JSX namespace, without which, typescript does not process jsx. This is also the same way react itself exports the JSX namespace